netmiko/nokia/nokia_sros_ssh.py

```diff
@@ -168,3 +170,95 @@ def cleanup(self, command="logout"):
+        """
+        Returns "//" if the current prompt is MD-CLI
+        empty string otherwise
+        """
+        return "//" if "@" in self.ssh_ctl_chan.base_prompt else ""
+
+    def remote_space_available(self, search_pattern=r"(\d+)\s+\w+\s+free"):
+        """Return space available on remote device."""
+
+        # Sample text for search_pattern.
+        # "               3 Dir(s)               961531904 bytes free."
+        self.ssh_ctl_chan.send_command(self._get_cmd_prefix() + "environment no more")
+        remote_cmd = self._get_cmd_prefix() + "file dir {}".format(self.file_system)
+        remote_output = self.ssh_ctl_chan.send_command(remote_cmd)
+        match = re.search(search_pattern, remote_output)
+        return int(match.group(1))
+
+    def check_file_exists(self, remote_cmd=""):
+        """Check if destination file exists (returns boolean)."""
+
+        if self.direction == "put":
+            self.ssh_ctl_chan.send_command(
+                self._get_cmd_prefix() + "environment no more"
+            )
+            if not remote_cmd:
+                remote_cmd = self._get_cmd_prefix() + "file dir {}/{}".format(
+                    self.file_system, self.dest_file
+                )
+            remote_out = self.ssh_ctl_chan.send_command(remote_cmd)
+            if "File Not Found" in remote_out:
+                return False
+            elif self.dest_file in remote_out:
+                return True
+            else:
+                raise ValueError("Unexpected output from check_file_exists")
+        elif self.direction == "get":
+            return os.path.exists(self.dest_file)
+
+    def remote_file_size(self, remote_cmd=None, remote_file=None):
+        """Get the file size of the remote file."""
+
+        if remote_file is None:
+            if self.direction == "put":
+                remote_file = self.dest_file
+            elif self.direction == "get":
+                remote_file = self.source_file
+        if not remote_cmd:
+            remote_cmd = self._get_cmd_prefix() + "file dir {}/{}".format(
+                self.file_system, remote_file
+            )
+        self.ssh_ctl_chan.send_command(self._get_cmd_prefix() + "environment no more")
+        remote_out = self.ssh_ctl_chan.send_command(remote_cmd)
+
+        if "File Not Found" in remote_out:
+            raise IOError("Unable to find file on remote system")
+
+        # Parse dir output for filename. Output format is:
+        # "10/16/2019  10:00p                6738 {filename}"
+
+        pattern = r"(\S+)\s+(\S+)\s+(\d+)\s+{}".format(re.escape(remote_file))
+        match = re.search(pattern, remote_out)
+
+        if not match:
+            raise ValueError("Filename entry not found in dir output")
+
+        file_size = int(match.group(3))
+        return file_size
+
+    def process_md5(self, md5_output, pattern=r"=\s+(\S+)"):
+        """ Nokia SROS does not support a md5sum calculation."""
+        pass
```

Comment:
Probably needs to raise an exception and then make a modification to the file_transfer function in scp_functions.py where if it is SR-OS, then disable_md5=True.

Basically, I think the current code would falsely give the impression that you were doing an MD5 when you weren't so we should make this explicit in the code.

Reply:
as discussed privately with you in the ntc slack, we will wait on your POC proposal that we can refine afterwards